File: GraphPlan Planner/src/edu/uno/ai/planning/graphplan/PlanGraphNode.java
package edu.uno.ai.planning.graphplan;

import java.util.List;

public interface PlanGraphNode {

	public int getInitialLevel();

	void setInitialLevel(int initialLevel);
	
	boolean existsAtLevel(int level);
	
	public List<?> getParentNodes();
	
	public List<?> getChildNodes();
	
}

File: UnitySokoban/Assets/Scripts/Planning/PlanGraph/PlanGraphNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PlanGraphProject
{
    public interface PlanGraphNode
    {
        int getInitialLevel();
        void setInitialLevel(int initialLevel);
        bool existsAtLevel(int level);
        List<PlanGraphNode> getParentNodes();
        List<PlanGraphNode> getChildNodes();
    }
}
